src/main/java/com/iexec/worker/tee/scone/TeeSconeService.java

```diff
@@ -45,112 +46,87 @@ public class TeeSconeService {
-            WorkerConfigurationService workerConfigService,
-            DockerService dockerService,
-            SgxService sgxService) {
-        this.sconeConfig = sconeConfig;
-        this.workerConfigService = workerConfigService;
-        this.dockerService = dockerService;
+            SmsClientProvider smsClientProvider,
+            SgxService sgxService,
+            TeeWorkflowConfigurationService teeWorkflowConfigurationService,
+            LasServicesManager lasServicesManager) {
+        super(smsClientProvider, teeWorkflowConfigurationService);
         this.sgxService = sgxService;
-        this.isLasStarted = sgxService.isSgxEnabled() && startLasService();
-        if (this.isLasStarted) {
+        this.lasServicesManager = lasServicesManager;
+
+        if (isTeeEnabled()) {
             log.info("Worker can run TEE tasks");
         } else {
             LoggingUtils.printHighlightedMessage("Worker will not run TEE tasks");
         }
     }
 
+    @Override
     public boolean isTeeEnabled() {
-        return isLasStarted;
+        return sgxService.isSgxEnabled();
     }
 
-    boolean startLasService() {
-        String lasImage = sconeConfig.getLasImageUri();
-        DockerRunRequest dockerRunRequest = DockerRunRequest.builder()
-                .containerName(sconeConfig.getLasContainerName())
-                .imageUri(lasImage)
-                // application & post-compose enclaves will be
-                // able to talk to the LAS via this network
-                .dockerNetwork(workerConfigService.getDockerNetworkName())
-                .sgxDriverMode(sgxService.getSgxDriverMode())
-                .maxExecutionTime(0)
-                .build();
-        if (!lasImage.contains(sconeConfig.getRegistryName())) {
-            throw new RuntimeException(String.format("LAS image (%s) is not " +
-                    "from a known registry (%s)", lasImage, sconeConfig.getRegistryName()));
-        }
-        DockerClientInstance client;
-        try {
-            client = dockerService.getClient(
-                    sconeConfig.getRegistryName(),
-                    sconeConfig.getRegistryUsername(),
-                    sconeConfig.getRegistryPassword());
-        } catch (Exception e) {
-            log.error("", e);
-            throw new RuntimeException("Failed to get Docker authenticated client to run LAS");
-        }
-        if (client == null) {
-            log.error("Docker client with credentials is required to enable TEE support");
-            return false;
-        }
-        if (!client.pullImage(lasImage)) {
-            log.error("Failed to download LAS image");
-            return false;
-        }
-
-        DockerRunResponse dockerRunResponse = dockerService.run(dockerRunRequest);
-        if (!dockerRunResponse.isSuccessful()) {
-            log.error("Failed to start LAS service");
-            return false;
-        }
-        return true;
+    @Override
```

Comment:
Pseudocode:
```suggestion
    @Override
    public Optional<ReplicateStatusCause> areTeePrerequisitesMetForTask(String chainTaskId){
        return super.areTeePrerequisitesMetForTask(chainTaskId) && prepareTeeForTask(chainTaskId);
    }
  

    @Override
```
(but also keep `prepareTeeForTask` call in `compute` stage as it is, in case the worker is restarted after `start` stage)

Reply:
Right, please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/464/commits/499432b5eabece882d2a2cd486879808be66c2df.

Reply:
Perfect thanks!